src/Page/Food.elm

```diff
@@ -0,0 +1,597 @@
+viewIngredient toString bar =
+    let
+        name =
+            bar.name |> processNameToString
+    in
+    div [ class "row align-items-center" ]
+        [ div [ class "col-sm-6 px-4 py-2 py-sm-0" ]
+            [ span [ class "d-block d-sm-none fs-7 text-muted" ] [ text "Quantité de l'ingrédient :" ]
+            , RangeSlider.ratio
+                { id = "slider-" ++ name
+                , update = IngredientSliderChanged bar.name
+                , value = bar.amount
+                , toString = toString
+                , disabled = not (isIngredient bar.name)
+                , min = 0
+                , max = 100
+                }
+            ]
+        , div [ class "col-sm-6" ]
+            [ barView bar
+            ]
+        ]
+
+
+type alias Bar =
+    { name : ProcessName
+    , amount : Unit.Ratio
+    , impact : Float
+    , width : Float
+    , percent : Float
+    }
+
+
+makeBar : Float -> Impact.Trigram -> ProcessName -> Process -> Bar
+makeBar totalImpact trigram processName process =
+    let
+        impact =
+            -- Product.getImpact trigram definitions impacts * Unit.ratioToFloat amount
+            Impact.grabImpactFloat Unit.PerItem Product.unusedDuration trigram process * Unit.ratioToFloat process.amount
+
+        percent =
+            impact * toFloat 100 / totalImpact
+    in
+    { name = processName
+    , amount = process.amount
+    , impact = impact
+    , width = clamp 0 100 percent
+    , percent = percent
+    }
+
+
+barView : Bar -> Html Msg
+barView bar =
+    div [ class "ps-3 py-1 border-top border-top-sm-0 border-start-0 border-start-sm d-flex" ]
+        [ div [ class "w-75 border my-2" ]
+            [ div
+                [ class "bg-primary"
+                , style "height" "100%"
+                , style "width" (String.fromFloat bar.width ++ "%")
+                ]
+                []
+            ]
+        , div [ class "text-end py-1 ps-2 text-truncate flex-fill" ]
+            [ Format.percent bar.percent ]
+        , div [ class "px-2 my-1" ] [ PieChart.view bar.percent ]
+        , button
+            [ class "btn"
+            , onClick <| DeleteIngredient bar.name
+            ]
+            [ i [ class "icon icon-trash" ] [] ]
+        ]
+
+
+floatToRoundedString : Int -> Float -> String
+floatToRoundedString exponent float =
+    float
+        |> Decimal.fromFloat
+        |> Decimal.roundTo exponent
+        |> Decimal.toStringIn Decimal.Dec
+
+
+ingredientSelector : (String -> Msg) -> List String -> Html Msg
+ingredientSelector event processes =
+    select
+        [ class "form-select"
+        , onInput event
+        ]
+        (option [] [ text "-- Sélectionner un ingrédient dans la liste --" ]
+            :: (processes
+                    |> List.map
+                        (\processName ->
+                            option [ value processName ] [ text processName ]
+                        )
+               )
+        )
```

Comment:
Nit: I know you're not a fan but this could also be written as

```elm
ingredientSelector : (String -> Msg) -> List String -> Html Msg
ingredientSelector event =
    List.map (\processName -> option [ value processName ] [ text processName ])
        >> (++) [ option [] [ text "-- Sélectionner un ingrédient dans la liste --" ] ]
        >> select [ class "form-select", onInput event ]
```